source/common/conn_pool/conn_pool_base.cc

```diff
@@ -425,19 +424,13 @@ void ConnPoolImplBase::checkForIdleAndCloseIdleConnsIfDraining() {
 
 void ConnPoolImplBase::onConnectionEvent(ActiveClient& client, absl::string_view failure_reason,
                                          Network::ConnectionEvent event) {
-  if (client.state() == ActiveClient::State::CONNECTING) {
-    ASSERT(connecting_stream_capacity_ >= client.effectiveConcurrentStreamLimit());
-    connecting_stream_capacity_ -= client.effectiveConcurrentStreamLimit();
```

Comment:
Can't we get a disconnect while the client was connecting?  if so I think we still need to update these.  if you agree please fix and add a regression test

Reply:
Yes, that possible. The accounting is updated in `if (event == Network::ConnectionEvent::RemoteClose ||
      event == Network::ConnectionEvent::LocalClose) {...}` block. This PR doesn't change this behavior, but moves it into each ConnectionEvent case.